### PR TITLE
feat(routing): add support for router tree state mocking

### PR DESCRIPTION
**Pull request by cgatian**
Closes: #328

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/ngneat/spectator/blob/master/CONTRIBUTING.md#commit
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
`firstChild`, `parent`, `children`, and `root` are not mockable by the `ActiveRoute`

Issue Number: #328

## What is the new behavior?
Adds additional properties to the `ActivatedRouteStub` to allow mocking portions of the router state tree.

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

### CODE REVIEW

File: projects/spectator/test/with-routing/my-page.component.spec.ts [hunk 1]
```diff
@@ -196,4 +196,65 @@ describe('MyPageComponent', () => {
...
+      firstChild: new ActivatedRouteStub({
+        params: { firstChild: '5' }
+      })
+    });
+
+    it('should support root mock', () => {
+      const spectator = createComponent();
+
+      let rootParam;
+      spectator.component.root!.paramMap.subscribe(params => {
```
Comment (NetanelBasal) [inline projects/spectator/test/with-routing/my-page.component.spec.ts:226]: Why not writing the expectations inside the `subscribe` block?

Comment (cgatian) [reply projects/spectator/test/with-routing/my-page.component.spec.ts:226]: Good point 👍

Comment (cgatian) [reply projects/spectator/test/with-routing/my-page.component.spec.ts:224]: Fixed

**Comment by NetanelBasal** [comment]
Add yourself to the contributors' list. Run `npm run contributors:add`

**Comment by cgatian** [comment]
@NetanelBasal updated 🎊

**Pull request merged**